Source/SessionManager/SessionManager.swift

```diff
@@ -35,7 +35,8 @@ public typealias LaunchOptions = [UIApplication.LaunchOptionsKey : Any]
 }
 
 @objc public protocol SessionActivationObserver: class {
-    func sessionManagerActivated(userSession : ZMUserSession)
+    func sessionManagerUserIsLoggedIn(isDatabaseLocked: Bool)
+    func sessionManagerRegisterDatabaseLocked(isDatabaseLocked: Bool)
```

Comment:
This approach looks like it should work. I would change the names though to still reflect that the session manager operates on user sessions. Maybe it even make sense to not distinguish between the the database locked/unlock case.

```suggestion
    func sessionManagerChangedActiveUserSession(isDatabaseLocked: Bool)
```